app/labor/cache_buster.rb

```diff
@@ -15,6 +15,7 @@ def self.bust(path)
                   headers: { "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"] })
   rescue URI::InvalidURIError => e
     Rails.logger.error("Trying to bust cache of an invalid uri: #{e}")
+    DataDogStatsClient.increment("cache_buster.invalid_uri")
```

Comment:
Given https://github.com/thepracticaldev/dev.to/blob/00c5a120eb314fbafc2d46b743673de78f9155f0/app/labor/cache_buster.rb#L10

I'm not sure if there's a great way to write a spec for this

Reply:
Adding `allow(Rails).to receive(:env) { "production".inquiry }` to the relevant spec should do the trick.

Reply:
Good idea! Now I'm wondering...what about Fastly, that's not set up with a dev environment, right? So if I add that to a spec and that block runs, isn't it going to attempt to _actually_ bust the cache?

Reply:
no, Fastly isn't setup in development, the cache buster also contains this line:

https://github.com/thepracticaldev/dev.to/blob/ca8ea0874d9ff7dab8a9c1efc727d673b72c5619/app/labor/cache_buster.rb#L9-L10

Reply:
Wait, on second thought, what do you mean @rhymes? That line you referenced is the same one I mentioned. So, in this case, are you saying we would **not** want to add

```ruby
allow(Rails).to receive(:env) { "production".inquiry }
```

because then the `POST` requests to Fastly would execute via the test suite and we don't want that, correct?
https://github.com/thepracticaldev/dev.to/blob/master/app/labor/cache_buster.rb#L9-L18

Reply:
`allow(HTTParty).to receice(:post).and_raise(URI::InvalidURIError)` if you want to avoid making requests should fix that, right @atsmith813?

Reply:
Ahhhhh I see - that makes a lot of sense. TIL about `StringInquirer`. Thank you so much! I think you're right, this isn't _really_ testing anything useful so I'll skip it 💯